capstone/src/main/java/com/google/sps/servlets/UpdateSeatsServlet.java

```diff
@@ -32,29 +41,32 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Long driverId = Long.parseLong(request.getParameter("driverId"));
-    Query query = new Query("Driver");
-    Driver res_driver = new Driver("","","","",0L,0L,0L);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    
+    Filter propertyFilter =
+        new FilterPredicate("id", FilterOperator.EQUAL, driverId);
+    Query q = new Query("Driver").setFilter(propertyFilter);
     List<Driver> drivers = new ArrayList<>();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(q);
     Long newSeats = 0L;
+    Driver res_driver = new Driver("","","","",0L,0L,0L);
     for (Entity entity : results.asIterable()) {  
```

Comment:
Does this handle the case where it can't find any driver? Overall I think the right thing to do would to be return a 404 status code.

Reply:
I will look into how to do this!

Reply:
It looks like calling asSingleEntity() is potentially an option since we expect exactly 0 or 1 results. There's also a way to get the number of results.

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asSingleEntity--

Reply:
The second part -- how to send a specific response status, would be something like this:

response.sendError(HttpServletResponse.SC_NOT_FOUND, "Driver not found");

Some additional documentation here: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html